src/System.Private.ServiceModel/src/System/IdentityModel/CryptoHelper.cs

```diff
@@ -61,7 +97,305 @@ internal static bool IsAsymmetricAlgorithm(string algorithm)
+                // For these algorithms above, crypto config returns internal objects.
+                // As we cannot create those internal objects, we are returning null.
+                // If no custom algorithm is plugged-in, at least these two algorithms
+                // will be inside the delegate dictionary.
+                case SecurityAlgorithms.Sha1Digest:
+#pragma warning disable CA5350 // Do not use insecure cryptographic algorithm SHA1.
+                    return SHA1.Create();
+#pragma warning restore CA5350 // Do not use insecure cryptographic algorithm SHA1.
+                case SecurityAlgorithms.ExclusiveC14n:
+                    throw ExceptionHelper.PlatformNotSupported();
+                case SHA256String:
+                case SecurityAlgorithms.Sha256Digest:
+                    return SHA256.Create();
+                case SecurityAlgorithms.Sha512Digest:
+                    return SHA512.Create();
+                case SecurityAlgorithms.Aes128Encryption:
+                case SecurityAlgorithms.Aes192Encryption:
+                case SecurityAlgorithms.Aes256Encryption:
+                case SecurityAlgorithms.Aes128KeyWrap:
+                case SecurityAlgorithms.Aes192KeyWrap:
+                case SecurityAlgorithms.Aes256KeyWrap:
+                    return Rijndael.Create();
+                case SecurityAlgorithms.TripleDesEncryption:
+                case SecurityAlgorithms.TripleDesKeyWrap:
+#pragma warning disable CA5350 // Do Not Use Weak Cryptographic Algorithms
+                    return TripleDES.Create();
+#pragma warning restore CA5350 // Do Not Use Weak Cryptographic Algorithms
+                case SecurityAlgorithms.HmacSha1Signature:
+                    byte[] key = new byte[64];
+                    new RNGCryptoServiceProvider().GetBytes(key);
+#pragma warning disable CA5350 // Do not use insecure cryptographic algorithm SHA1.
+                    return new HMACSHA1(key);
+#pragma warning restore CA5350 // Do not use insecure cryptographic algorithm SHA1.
+                case SecurityAlgorithms.HmacSha256Signature:
+                    return new HMACSHA256();
+                case SecurityAlgorithms.ExclusiveC14nWithComments:
+                    throw ExceptionHelper.PlatformNotSupported();
+                case SecurityAlgorithms.Ripemd160Digest:
+                    return null;
+                case SecurityAlgorithms.DesEncryption:
+#pragma warning disable CA5351 // Do Not Use Broken Cryptographic Algorithms
+                    return DES.Create();
+#pragma warning restore CA5351 // Do Not Use Broken Cryptographic Algorithms
+                default:
+                    return null;
+            }
+        }
+
+
+        internal static object GetAlgorithmFromConfig(string algorithm)
+        {
+            if (string.IsNullOrEmpty(algorithm))
+            {
+                throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(new ArgumentNullException("algorithm"));
```

Comment:
Is dotnet/wcf not using `nameof()`? => `... new ArgumentNullException(nameof(algorithm))`

Reply:
I fixed as many as I could do with regex find/replace but I didn't get them all. Do you know of a Roslyn analyzer which can fix this through an entire project?

---
In reply to: [274001822](https://github.com/dotnet/wcf/pull/3538#discussion_r274001822) [](ancestors = 274001822)

Reply:
I just realized I had fixed this in the second commit which has all the cleanup such as what you mentioned.

---
In reply to: [274725688](https://github.com/dotnet/wcf/pull/3538#discussion_r274725688) [](ancestors = 274725688,274001822)